static/js/containers/Payment_test.js

```diff
@@ -1,90 +1,186 @@
+/* global SETTINGS: false */
 import React from 'react';
 import configureTestStore from 'redux-asserts';
 import { mount } from 'enzyme';
 import { Provider } from 'react-redux';
 import { assert } from 'chai';
 import sinon from 'sinon';
+import _ from 'lodash';
 
 import * as api from '../lib/api';
-import { setTotal } from '../actions';
+import {
+  setPaymentAmount,
+  setSelectedKlassIndex
+} from '../actions';
 import rootReducer from '../reducers';
 import Payment from '../containers/Payment';
 import * as util from '../util/util';
-import { makeRequestActionType, makeReceiveSuccessActionType } from '../rest';
+import {
+  makeRequestActionType,
+  makeReceiveSuccessActionType
+} from '../rest';
 
 const REQUEST_PAYMENT = makeRequestActionType('payment');
 const RECEIVE_PAYMENT_SUCCESS = makeReceiveSuccessActionType('payment');
+const REQUEST_KLASSES = makeRequestActionType('klasses');
+const RECEIVE_KLASSES_SUCCESS = makeReceiveSuccessActionType('klasses');
+
+const generateFakeKlasses = (numKlasses = 1) => {
+  return _.times(numKlasses, (i) => ({
+    klass_name: `Bootcamp 1 Klass ${i}`,
+    klass_id: i
+  }));
+};
+
+describe('Payment container', () => {
+  let store, listenForActions, sandbox, fetchStub,
+    klassesUrl, klassesStub; // eslint-disable-line no-unused-vars
 
-describe('Payment', () => {
-  let store, listenForActions, sandbox;
   beforeEach(() => {
+    SETTINGS.user = {
+      full_name: "john doe",
+      username: "johndoe"
+    };
+
     store = configureTestStore(rootReducer);
-    sandbox = sinon.sandbox.create();
     listenForActions = store.createListenForActions();
+    sandbox = sinon.sandbox.create();
+    fetchStub = sandbox.stub(api, 'fetchJSONWithCSRF');
+    klassesUrl = `/api/v0/klasses/${SETTINGS.user.username}/`;
   });
+
   afterEach(() => {
     sandbox.restore();
   });
 
-  let renderPayment = () => {
-    return mount(
+  let renderPaymentComponent = (props = {}) => (
+    mount(
       <Provider store={store}>
-        <Payment />
+        <Payment {...props} />
       </Provider>
-    );
+    )
+  );
+
+  let renderFullPaymentPage = (props = {}) => {
+    let wrapper;
+    return listenForActions([REQUEST_KLASSES, RECEIVE_KLASSES_SUCCESS], () => {
+      wrapper = renderPaymentComponent(props);
+    }).then(() => {
+      return Promise.resolve(wrapper);
+    });
   };
 
-  it('sets a price', () => {
-    let payment = renderPayment();
-    payment.find('input[id="total"]').props().onChange({
-      target: {
-        value: "123"
-      }
+  it('sets a selected klass', () => {
+    let fakeKlasses = generateFakeKlasses(3);
+    klassesStub = fetchStub.withArgs(klassesUrl)
```

Comment:
can you move this setup (with `klassesStub`) into the `beforeEach` function? seems like the same setup is happening in a few functions.